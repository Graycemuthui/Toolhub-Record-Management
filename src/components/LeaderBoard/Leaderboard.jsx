import { Box, Typography, useTheme } from "@mui/material";
import { DataGrid } from "@mui/x-data-grid";
import { tokens } from "../../color";
import { records } from "../data/MockData";

const Leaderboard = () => {
  const theme = useTheme();
  const colors = tokens(theme.palette.mode);
  const columns = [
    {
      field: "name",
      headerName: "Name",
      flex: 1,
      cellClassName: "name-column--cell",
    },
    {
      field: "edits",
      headerName: "Edits",
      type: "number",
      flex: 1,
    },
  ];
  return (
    <Box m="20px">
      <Typography title="TOTAL" subtitle="Total number of tool records" />
      <Box
        m="100px 0 0 0"
        height="75vh"
        sx={{
          "& .MuiDataGrid-root": {
            border: "none",
          },
          "& .MuiDataGrid-cell": {
            borderBottom: "none",
          },
          "& .name-column--cell": {
            color: colors.greenAccent[400],
          },
          "& .MuiDataGrid-columnHeaders": {
            backgroundColor: colors.primary[900],
            borderBottom: "none",
          },
          "& .MuiDataGrid-virtualScroller": {
            backgroundColor: colors.primary[400],
          },
          "& .MuiDataGrid-footerContainer": {
            borderTop: "none",
            backgroundColor: colors.primary[900],
          },
          "& .MuiCheckbox-root": {
            color: `${colors.grey[200]} !important`,
          },
        }}
      >
        <DataGrid checkboxSelection rows={records} columns={columns} />
      </Box>
    </Box>
  );
};

export default Leaderboard;
