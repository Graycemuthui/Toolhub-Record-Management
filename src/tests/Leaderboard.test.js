import React from "react";
import TestRenderer from "react-test-renderer";
import Leaderboard from "../../src/components/LeaderBoard/Leaderboard";

it("Leaderboard render correctly", () => {
  const tree = TestRenderer.create(<Leaderboard />).toJSON();
  expect(tree).toMatchSnapshot();
});
