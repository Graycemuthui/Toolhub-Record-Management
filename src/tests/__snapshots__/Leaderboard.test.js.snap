// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard render correctly 1`] = `
<div
  className="MuiBox-root css-zwiamn"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    subtitle="Total number of tool records"
    title="TOTAL"
  />
  <div
    className="MuiBox-root css-uoqp5g"
  >
    <div
      aria-colcount={3}
      aria-multiselectable={false}
      aria-rowcount={19}
      className="MuiDataGrid-root MuiDataGrid-root--densityStandard css-1e2bxag-MuiDataGrid-root"
      role="grid"
    >
      <div
        className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          className="MuiDataGrid-overlay css-f7tavo-MuiDataGrid-overlay"
          error={[TypeError: Cannot set properties of null (setting 'scrollLeft')]}
        >
          An error occurred.
        </div>
      </div>
    </div>
  </div>
</div>
`;
